File: solutions/python/allergies/1/allergies.py
class Allergies(object):
    score = 0
    allergy_list = ['eggs', 'peanuts', 'shellfish', 'strawberries', 'tomatoes', 'chocolate', 'pollen', 'cats']
    my_allergies = []

    def __init__(self, score):
        self.score = score
        self.my_allergies=[]
        s = score
        #print ("---->s=",s)
        for i in range(0,8):
            r = s % 2
            s = s // 2;
            if r == 1:
                self.my_allergies.append(self.allergy_list[i])
        #print(self.my_allergies)


    def is_allergic_to(self, item):
        return item in set(self.my_allergies)

    @property
    def lst(self):
        return self.my_allergies

File: solutions/javascript/allergies/1/allergies.js
//
// This is only a SKELETON file for the 'Allergies' exercise. It's been provided as a
// convenience to get you started writing code faster.
//
const ALNAMES= ['eggs', 'peanuts', 'shellfish','strawberries', 'tomatoes','chocolate','pollen','cats'];

export class Allergies {
  constructor(aln) {
    this.aln = aln;
  }

  list() {
    var i=[];
    var n = this.aln;
    for (var j=0; j <= 7 ; j++) {
      i.push(n % 2);
      n = Math.floor(n / 2);
    }
    var alleg = [];
    for (var j=0; j <= 7 ; j++) {
      if (i[j] == 1) alleg.push(ALNAMES[j]);
    }
    return alleg;
  }

  allergicTo(alname) {
    if (this.list().indexOf(alname)>=0)
      return true;
    return false;
  }
};
